Mathlib/Analysis/Calculus/PartialDerivatives/Notation.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright (c) 2025 Paul Lezeau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Lezeau, Eric Wieser
+-/
+import Mathlib.Analysis.Calculus.LineDeriv.Basic
+import Mathlib.LinearAlgebra.Basis.HasCanonicalBasis
+
+
+/-! # Notation for partial derivatives
+
+This file introduces some custom notation for partial derivatives. In order to make this
+as flexible as possible (e.g. we would like to have consistent notation that works for
+`ℝ × ℝ`, `Fin 2 → ℝ` and `EuclideanSpace ℝ (Fin 2)`), we work on abstract vector spaces
+equiped with a `HasCanonicalBasis` instance.
+
+We introduce two different notations
+- Whenever the index `i` is subscriptable, we can use `∂ᵢ[ℝ] f` to denote the `i-th` partial
+  derivative. This works for e.g `i : Fin n`.
+- In general, if the basis is indexed by a type `ι`, then we can use `∂[i; ℝ] f` for the
+  `i`-th partial derivative.
+
+-/
+
+universe u v w u' v' w'
+
+namespace HasCanonicalBasis
+/--
+The partial derivative of a function `F : V → 𝕜`.
+
+`partialDeriv 𝕜 F i tx` return the partial derivative of function `F : V → 𝕜`
+with respect to the `i`-th component of the canonical basis of `V` at point `tx : V`.
+
+This can be written using the following notations:
+- `(∂ᵢ[ℝ] f) tx` if the index `i` is subscriptable,
+- `(∂[i;ℝ] f) tx` in general.
+-/
+@[simp, nolint unusedArguments] -- we need `HasCanonicalBasis 𝕜 V ι f` to make the notation work
+noncomputable def partialDeriv (𝕜 : Type u) {V : Type v} {ι : Type w} {E : Type*}
```

Comment:
I think this should be `abbrev` instead of `partialDeriv` otherwise you will have to duplicate whole API for `lineDeriv` or start every proof with `unfold partialDeriv`.